haystack-commons/src/main/java/com/expedia/www/haystack/metrics/HaystackGraphiteNamingConvention.java

```diff
@@ -0,0 +1,67 @@
+package com.expedia.www.haystack.metrics;
+
+import com.netflix.servo.Metric;
+import com.netflix.servo.annotations.DataSourceType;
+import com.netflix.servo.monitor.MonitorConfig;
+import com.netflix.servo.publish.graphite.GraphiteNamingConvention;
+import com.netflix.servo.tag.Tag;
+import com.netflix.servo.tag.TagList;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static com.expedia.www.haystack.metrics.MetricObjects.TAG_KEY_CLASS;
+import static com.expedia.www.haystack.metrics.MetricObjects.TAG_KEY_SUBSYSTEM;
+
+public class HaystackGraphiteNamingConvention implements GraphiteNamingConvention {
+    static final String HOST_NAME_UNKNOWN_HOST_EXCEPTION = "HostName-UnknownHostException";
+    static final String MISSING_TAG = "MISSING_TAG_%s";
+    static final String MISSING_VALUE = "MISSING_VALUE_%s";
+    static final String METRIC_FORMAT = "%s.%s.%s.%s_%s";
+
+    static Factory factory = new Factory(); // will be mocked out in unit tests
+    static final String LOCAL_HOST_NAME = getLocalHost();
+
+    static String getLocalHost() {
+        try {
+            return cleanup(factory.getLocalHost().getHostName(), "");
+        } catch (UnknownHostException e) {
+            return HOST_NAME_UNKNOWN_HOST_EXCEPTION;
+        }
+    }
+
+    @Override
+    public String getName(Metric metric) {
+        final MonitorConfig config = metric.getConfig();
+        final TagList tags = config.getTags();
+        final String subsystem = cleanup(tags, TAG_KEY_SUBSYSTEM);
+        final String klass = cleanup(tags, TAG_KEY_CLASS);
+        final String configName = config.getName(); // Servo disallows null for name, no need to cleanup
+        final String type = cleanup(tags, DataSourceType.KEY);
+        return String.format(METRIC_FORMAT, LOCAL_HOST_NAME, subsystem, klass, configName, type);
```

Comment:
type is the metric name created by Servo (RATE in the case of a Counter). I don't mind switching the two strings, but it will mean a change to the filter in the InfluxDb TOML file (to create the tags properly). Do you still want me to make this change?

Reply:
Yes.  Though we know we use Influx ( and hence these are broken into tags ), it is better to not assume that in this code and keep it agnostic of who ingests this graphite compatible data.  If "real" graphite consumes the metric, presence of host name will make it hard to aggregate. It will be easier to say ```haystack-trends.*``` than ```*.haystack-trends```

Reply:
Change made.